### PR TITLE
chore: update Readme and Contributing guide

### PR DESCRIPTION
**Pull request by bhongy**
- Make the repo description and the packages installation guide clearer
- Move contributing related instructions (e.g. setting up local development environment and releasing a new version) to `CONTRIBUTING.md`
- Update "Opening a Pull Request" and "Releasing a new version" instruction to make it up-to-date with https://github.com/formspree/formspree-js/pull/39
- Move `CONTRIBUTING.md` from `.github` folder to root
- Clarify "Setting up your local repo" instruction
- Add list of development commands (test, typecheck)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,41 @@
 # Formspree JS
 
-Monorepo containing the @formspree/core and @formspree/react packages. Built with [turborepo](https://turbo.build/).
+A monorepo containing libraries for seamless form integration with [Formspree](https://formspree.io/) via Javascript and/or React.
```
Comment (bhongy) [inline README.md:3]: @colevscode any feedback on this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+_Note: non-packages (examples/\*) do not need changesets._
+
+## Releasing a new version
+
+_Note: Only core maintainers can release new versions of formpree-js packages._
+
+When we are ready to release a new version on one or more packages, follow these steps:
+
+1. Create a new branch off the `main` branch.
+1. Run `yarn changeset version` which will consume the changeset files and update the changelog and package.json files in the relevant packages.
```
Comment (bhongy) [inline CONTRIBUTING.md:58]: This is not quite accurate. Once a PR with a changeset is merged to `main`, our `release` GH workflow will create a "Version Packages" PR automatically ([example release run](https://github.com/formspree/formspree-js/actions/runs/5353146416/jobs/9708747974), [example PR](https://github.com/formspree/formspree-js/pull/40)).

This needs to be updated.

Comment (bhongy) [reply CONTRIBUTING.md]: Updated in c57537a

File: README.md (hunk 1 above)
Comment (colevscode) [reply README.md:3]: 👍

**Review by colevscode — APPROVED** [review]

**Pull request merged**